workflow-runtime/src/commonTest/kotlin/com/squareup/workflow1/WorkflowOperatorsTest.kt

```diff
@@ -207,23 +207,37 @@ class WorkflowOperatorsTest {
   private abstract class StateFlowWorkflow<T>(
     val name: String,
     val flow: StateFlow<T>
-  ) : StatelessWorkflow<Unit, Nothing, T>() {
+  ) : StatefulWorkflow<Unit, T, Nothing, T>() {
     var starts: Int = 0
       private set
 
+    override fun initialState(
+      props: Unit,
+      snapshot: Snapshot?
+    ): T {
+      return flow.value
+    }
+
     private val rerenderWorker = object : Worker<T> {
       override fun run(): Flow<T> = flow.onStart { starts++ }
     }
 
     override fun render(
       renderProps: Unit,
+      renderState: T,
       context: RenderContext
     ): T {
       // Listen to the flow to trigger a re-render when it updates.
-      context.runningWorker(rerenderWorker as Worker<Any?>) { WorkflowAction.noAction() }
-      return flow.value
+      context.runningWorker(rerenderWorker) { output: T ->
+        action("rerenderUpdate") {
+          state = output
```

Comment:
prior to this state was not actually changing - so this would not work in the optimized runtime.